mypy/semanal.py

```diff
@@ -3120,6 +3042,25 @@ def builtin_type(self, fully_qualified_name: str) -> Instance:
         assert isinstance(node, TypeInfo)
         return Instance(node, [AnyType(TypeOfAny.special_form)] * len(node.defn.type_vars))
 
+    def add_builtin_aliases(self, tree: MypyFile) -> None:
+        # For historical reasons, the aliases like `List = list` are not defined
```

Comment:
We could just define those in typeshed actually.

Reply:
I would keep this out of this PR. Will pytype an other be OK with this? (There are potential problems with circular dependencies). If yes we can easily do this later, fortunately with this refactoring it is just a matter of removing this patching step (before `normalize_type_alias` was all over the `semanal` code).

Reply:
I believe other type checkers don't use typing.pyi from typeshed; they have their own versions.

But I agree that this PR is doing enough things already. :)